students/Harry_Maher/Session04/_filepractice.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""
+    Write a program which prints the full path for all files in the current directory, one per line
+    Write a program which copies a file from a source, to a destination (without using shutil, or the OS copy command).
+        Advanced: make it work for any size file: i.e. don’t read the entire contents of the file into memory at once.
+        This should work for any kind of file, so you need to open the files in binary mode: open(filename, 'rb') (or 'wb' for writing). Note that for binary files, you can’t use readline() – lines don’t have any meaning for binary files.
+        Test it with both text and binary files (maybe jpeg or something of your choosing).
+"""
+import os
+
+def print_all_files():
+    for file in os.listdir():
+        print(os.getcwd()+"/"+file) #though windows uses "\" for some reason sometimes?
+ 
```

Comment:
See this stackoverflow article for a way to make this platform agnostic: 

https://stackoverflow.com/questions/6036129/platform-independent-file-paths

Reply:
Ahh thanks!